src/Microsoft.AspNetCore.SignalR.Client.Core/HubConnection.cs

```diff
@@ -275,15 +287,19 @@ private Task Shutdown(Exception ex = null)
         private Task DispatchInvocationAsync(InvocationMessage invocation, CancellationToken cancellationToken)
         {
             // Find the handler
-            if (!_handlers.TryGetValue(invocation.Target, out InvocationHandler handler))
+            if (!_handlers.TryGetValue(invocation.Target, out List<InvocationHandler> handlers))
             {
                 _logger.MissingHandler(invocation.Target);
                 return Task.CompletedTask;
             }
-
+            var tasks = new List<Task>();
+            foreach(var handler in handlers)
```

Comment:
Isn't it super racey? The list could be modified if someone happens to be adding or removing a handler while a message is received.